src/modm/driver/can/mcp2515.hpp

```diff
@@ -186,28 +209,61 @@ namespace modm
 			BIT_MODIFY = 0x05
 		};
 
+		modm::ResumableResult<bool>
+		mcp2515ReadMessage();
+
+		bool
+		mcp2515IsReadyToSend(uint8_t status);
+
+		modm::ResumableResult<bool>
+		mcp2515IsReadyToSend();
+
+		modm::ResumableResult<bool>
+		mcp2515SendMessage(const can::Message& message);
+
 		static void
 		writeRegister(uint8_t address, uint8_t data);
 
 		static uint8_t
 		readRegister(uint8_t address);
 
-		static void
+		void
 		bitModify(uint8_t address, uint8_t mask, uint8_t data);
 
-		static uint8_t
+		modm::ResumableResult<uint8_t>
 		readStatus(uint8_t type);
 
-		static inline void
+		modm::ResumableResult<void>
 		writeIdentifier(const uint32_t& identifier, bool isExtendedFrame);
 
-		static inline bool
+		modm::ResumableResult<bool>
 		readIdentifier(uint32_t& identifier);
 
-	protected:
+	private:
+		inline static modm::atomic::Queue<modm::can::Message, 32> txQueue;
+		inline static modm::atomic::Queue<modm::can::Message, 32> rxQueue;
+
 		static SPI spi;
 		static CS chipSelect;
 		static INT interruptPin;
+
+		modm::can::Message messageBuffer_;
+		modm::ShortTimeout delayR_;
+		modm::ShortTimeout delayS_;
+		uint8_t statusBuffer_ = 0;
+		uint8_t statusBufferR_ = 0;
+		uint8_t statusBufferS_ = 0;
+		uint8_t statusBufferReady_ = 0;
+		uint8_t addressBufferR_ = 0;
+		uint8_t addressBufferS_ = 0;
+		uint8_t i_, j_ = 0;
+		uint8_t a_ = 0;
+		uint8_t b_ = 0;
+		uint8_t data_ = 0;
+		bool tempR_ = false;
+		bool tempS_ = false;
+		bool temp_ = false;
+		bool receiveSuccess_ = false;
```

Comment:
Postfixing class members with `_` is not usual modm code style.

Reply:
well hmk